src/prefect/flows.py

```diff
@@ -808,8 +815,15 @@ async def from_source(
         """
         if isinstance(source, str):
             storage = create_storage_from_url(source)
-        else:
+        elif isinstance(source, RunnerStorage):
             storage = source
+        elif hasattr(source, "get_directory"):
+            storage = BlockStorageAdapter(source)
```

Comment:
I'm curious if there's a reason to prefer a `hasattr` check as opposed to doing an `isinstance(source, ReadableDeploymentStorage)` -- are there cases where `get_directory` wouldn't be implemented?

Reply:
Yeah, we have a `WritableDeploymentStorage` class that has a `get_directory` method, but doesn't inherit from `ReadableDeploymentStorage`, so I opted to check for the method directly. I could check for both `DeploymentStorage` types instead, but I only need that one method.

Reply:
Any thought about checking for the `get-directory` capability?  Could make it possible to use more kinds of blocks in the future?  I guess they'd all have to have a `get_directory` method anyway in that case...

Reply:
Block capabilities are a proxy for determining what methods a `Block` subclass has when you don't have have the actual class (e.g. server side). Since we'll have access to a Python object, we can inspect the object directly for a higher degree of confidence that the provided `Block` subclass will work.